.circleci/config.yml

```diff
@@ -2,32 +2,38 @@
 # Reference doc: https://circleci.com/docs/2.0/project-walkthrough/
 version: 2.1
 jobs:
-  build_test_jvm:
+  build_test_jvm_integration:
     docker:
-      - image: circleci/openjdk:8u242-jdk
+      - image: ducouloa/ml4ir:0.1
+    resource_class: xlarge
     steps:
       - checkout
       - restore_cache:
-          key: maven-deps-{{ checksum "jvm/pom.xml" }}
+          key: maven-python-deps-{{ checksum "jvm/pom.xml" }}-{{ checksum "python/requirements.txt" }}
+      - run:
+          name: Install Python deps in a venv
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install -r python/requirements.txt
       - run:
           name: Build
           command: |
             cd jvm
             mvn clean -Dmaven.test.skip=true install
       - save_cache:
-          key: maven-deps-{{ checksum "jvm/pom.xml" }}
+          key: maven-python-deps-{{ checksum "jvm/pom.xml" }}-{{ checksum "python/requirements.txt" }}
           paths:
             - ~/.m2
+            - "venv"
       - run:
-          name: Run JVM tests
+          name: Run JVM tests and integration tests
           command: |
+            . venv/bin/activate
             cd jvm
-            mvn test
-            mvn surefire-report:report-only
-      - store_artifacts:
-          path: jvm/ml4ir-inference/target/site/surefire-report.html
+            mvn verify -Pintegration_tests
       - store_test_results:
-          path: jvm/ml4ir-inference/target/site/surefire-report.html
+          path: jvm/ml4ir-inference/target
```

Comment:
we keep the entire target dir now?  why?

Reply:
1. The path to the report was not good, it doesn't exist (if you look at previous jobs, you would see no test results aggregate)
2. Putting the target directory allows to aggregate the results of surefire (utest) and failsafe (itest) at once.
3. It's CircleCi compatible to point to a "root" result directory, and circleCi infers the test suite name from the generic result directory names.